### ISSUE TITLE
The start scripts dirty vdevs when importing from the cachefile

### ISSUE THREAD

**Issue by ryao** [openzfs/zfs#3434]
@FransUrbo @behlendorf As I mentioned in #3431, if the start scripts import a pool, they will dirty the vdev labels, causing an update. This is a regression from the Illumos behavior (that we also support) where the modules load the pool from the cachefile without dirtying the labels. It is problematic because dirtying the labels triggers a label update that increments the stored txg since the last label update, which prevents txg rollback when something goes wrong. Dirtying the labels is only necessary when the vdev tree is changed. If we are going to have the scripts do an import from the cachefile, it should be a verbatim import to be consistent with the Illumos behavior.

**Comment by FransUrbo** [comment]
Why is this happening? The title indicates that there's something wrong with the init/initrd scripts, but I've checked and <code>zpool import</code> is used correctly according to the documentation.

**Comment by ryao** [comment]
The documentation is meant for system administrators' use of the tools, not system initialization scripts. The `-F` switch is intentionally undocumented because it is meant to be used only by system tools. The change in the label's txg number (not the most recent uberblock's txg) breaks rollback if something goes wrong and is something that is only done on Illumos when a vdev configuration change is made or pool is manually imported. The system initialization scripts should not import the pool in the same way that a system administrator would.

**Comment by behlendorf** [comment]
Now I see what you're getting at.  But this should only be the case when the cache file is stale with respect to the pool.  This is something which is explicitly [checked for](https://github.com/zfsonlinux/zfs/blob/master/module/zfs/spa.c#L2807) in spa_import(), your just forcing this behavior by making it a verbatim import.

So why is the cache file stale?  I can see how this could be the case for a cache file which is stored an initramfs and therefore never gets updated.  But to handle that case I think we should probably investigate reinstating `spa_import_rootpool()` which is designed for exactly this situation.  We could certainly add an option to `zpool import` for this.

I'm also not a fan of depending on an undocumented developer option for the core boot process.

**Comment by ryao** [comment]
I am for any solution that avoids dirtying the vdev labels during normal boot.

**Comment by behlendorf** [comment]
@ryao have you seen this occur for any case other than a stale cache file (presumably in an initramfs)?

**Comment by ryao** [comment]
@behlendorf Yes. I found it when debugging the following flocker bug on 1834:

https://clusterhq.atlassian.net/browse/FLOC-1834

I observed that the vdev labels of a a file-backed pool on CentOS 7 reliably were dirtied on every system boot. That is a bug, although not the bug that I was investigating. However, it is related to other bugs that I had investigated in the past with only partial luck, such as #2452.
